include/nuttx/net/net.h

```diff
@@ -253,6 +253,26 @@ struct socket
   FAR const struct sock_intf_s *s_sockif;
 };
 
+struct socket_ctx_data_s
+{
+  uint8_t s_domain;  /* IP domain */
+  uint8_t s_type;    /* Protocol type */
+  uint8_t s_proto;   /* Socket Protocol */
+  uint8_t s_priv[0]; /* Private socket context */
+};
+
+/* NETDEV ioctl command:
+ *
+ * Command:      SIOCGETCONTEXT or SIOCSETCONTEXT
+ * Description:  Get or Set socket context.
+ */
+
+struct socket_context_s
+{
+  size_t ctx_size;               /* Size of socket context */
+  struct socket_ctx_data_s *ctx; /* Buffer where socket context is stored */
```

Comment:
should we embed socket_ctx_data_s inside socket_context_s directly? otherwise, it's very hard to pass the pointer to usersock daemon in a general way.

Reply:
I think this implementation works for me, but is there any reason why it can't be a pointer?
I have confirmed that the usrsock daemon can receive the pointer specified by the application. (I have not modified the usrsock implementation.)

Reply:
you need consider kernel mode, it's impossible to pass the pointer cross the process boundary.